frontend/src/utils/prepareLocalStorageData.ts

```diff
@@ -0,0 +1,62 @@
+import getMostRecentPastTerm, { MostRecentTerm } from 'utils/getMostRecentPastTerm';
+import { parseMarkToInt } from 'pages/TermPlanner/utils';
+import { DegreeSliceState } from 'reducers/degreeSlice';
+import { PlannerSliceState } from 'reducers/plannerSlice';
+
+type TermPlan = {
+  // key = course code, value = [UOC, mark]
+  [courseCode: string]: [number, number | null]
+};
+
+type YearPlan = TermPlan[];
+
+type LocalStorageData = {
+  programCode: string
+  specialisations: string[]
+  year: number
+  plan: YearPlan[]
+  mostRecentPastTerm: MostRecentTerm
+  programName: string
+  startYear: number
+  numYears: number
+  unplanned: string[]
+  isSummerEnabled: boolean
+};
+
+const prepareLocalStorageData = (
+  planner: PlannerSliceState,
+  degree: DegreeSliceState,
+): LocalStorageData => {
+  const {
+    years, startYear, courses, unplanned, numYears, isSummerEnabled,
+  } = planner;
+  const { programCode, programName, specs } = degree;
+
+  const plan: YearPlan[] = [];
```

Comment:
we are basically copyiong the logic when creating the plan from the other prepareCoursesForValidation util function right? Do you expect the two endpoints require the same schema for the plan? If so, might be better to abstract it out into another util function like `createYearPlane` that creates the year plan and just use that util function in both places?